contracts/strategies/ltip-simple/LTIPSimpleStrategy.sol

```diff
@@ -0,0 +1,708 @@
+    /// @custom:data-registry (address recipientId, address recipientAddress, uint256 allocationAmount, Metadata metadata)
+    /// @custom:data-no-registry (address recipientAddress, address registryAnchor, uint256 allocationAmount, Metadata metadata)
+    /// @param _sender The sender of the transaction
+    /// @return recipientId The id of the recipient
+    function _registerRecipient(bytes memory _data, address _sender)
+        internal
+        override
+        onlyActiveRegistration
+        returns (address recipientId)
+    {
+        bool isUsingRegistryAnchor;
+        address recipientAddress;
+        address registryAnchor;
+        uint256 allocationAmount;
+        Metadata memory metadata;
+
+        // decode data custom to this strategy
+        if (registryGating) {
+            (recipientId, recipientAddress, allocationAmount, metadata) =
+                abi.decode(_data, (address, address, uint256, Metadata));
+
+            // when registry gating is enabled, the recipientId must be a profile member
+            if (!_isProfileMember(recipientId, _sender)) revert UNAUTHORIZED();
+        } else {
+            (recipientAddress, registryAnchor, allocationAmount, metadata) =
+                abi.decode(_data, (address, address, uint256, Metadata));
+            isUsingRegistryAnchor = registryAnchor != address(0);
+            recipientId = isUsingRegistryAnchor ? registryAnchor : _sender;
+
+            // when using registry anchor, the ID of the recipient must be a profile member
+            if (isUsingRegistryAnchor && !_isProfileMember(recipientId, _sender)) revert UNAUTHORIZED();
+        }
+
+        // Check if the metadata is required and if it is, check if it is valid, otherwise revert
+        if (metadataRequired && (bytes(metadata.pointer).length == 0 || metadata.protocol == 0)) {
+            revert INVALID_METADATA();
+        }
+
+        // If the recipient address is the zero address this will revert
+        if (recipientAddress == address(0)) revert RECIPIENT_ERROR(recipientId);
+
+        // Get the recipient
+        Recipient storage recipient = _recipients[recipientId];
+
+        // update the recipients data
+        recipient.recipientAddress = recipientAddress;
+        recipient.useRegistryAnchor = registryGating ? true : isUsingRegistryAnchor;
+        recipient.allocationAmount = allocationAmount;
+        recipient.metadata = metadata;
+        recipient.recipientStatus = Status.Pending;
+
+        if (recipient.recipientStatus == Status.None) {
+            // If the recipient status is 'None' add the recipient to the '_recipientIds' array
+            _recipientIds.push(recipientId);
+            emit Registered(recipientId, _data, _sender);
+        } else {
+            emit UpdatedRegistration(recipientId, _data, _sender);
+        }
```

Comment:
Since you set `recipient.recipientStatus = Status.Pending` this will never be `Status.None` right ?

Reply:
Good catch, sorry

Reply:
We can do two things, I can check in this flow if at the start of the function call the status === NONE, and emit a Registered event or RegistrationUpdated event according to that check. 

Or add an 'updateApplication' flow, which would mostly overlap with this method but allow for updating [recipientAddress, allocationAmount, metadata].

Thoughts?

Reply:
I dont think there is a need for the `updateApplication` flow, just moving the `if-else` under the recipient error is good enough https://github.com/allo-protocol/allo-v2/blob/037977ec9bf5b5499acc50cb838c02e2b26066cd/contracts/strategies/ltip-simple/LTIPSimpleStrategy.sol#L512